### PR TITLE
destination-mysql: convert prod code to kotlin

### PR DESCRIPTION
**Pull request by stephane-airbyte**
## Can this PR be safely reverted and rolled back?

- [ ] YES 💚
- [ ] NO ❌

### CODE REVIEW

File: airbyte-integrations/connectors/destination-mysql-strict-encrypt/src/main/java/io/airbyte/integrations/destination/mysql/MySQLDestinationStrictEncrypt.java [hunk 1]
```diff
@@ -11,6 +11,7 @@
 import io.airbyte.cdk.integrations.base.spec_modification.SpecModifyingDestination;
 import io.airbyte.commons.json.Jsons;
 import io.airbyte.protocol.models.v0.ConnectorSpecification;
+import io.airbyte.integrations.destination.mysql.MySQLDestination;
```
Comment (edgao) [inline airbyte-integrations/connectors/destination-mysql-strict-encrypt/src/main/java/io/airbyte/integrations/destination/mysql/MySQLDestinationStrictEncrypt.java:14]: why are these new imports necessary?

File: airbyte-integrations/connectors/destination-mysql/src/testFixtures/kotlin/io/airbyte/integrations/destination/mysql/MysqlContainerFactory.kt [hunk 2]
```diff
@@ -11,6 +11,6 @@ import org.testcontainers.utility.DockerImageName
 /** Much like the destination-postgres PostgresTestDatabase, this was copied from source-mysql. */
 class MySQLContainerFactory : ContainerFactory<MySQLContainer<*>>() {
     override fun createNewContainer(imageName: DockerImageName?): MySQLContainer<*> {
-        return MySQLContainer(imageName?.asCompatibleSubstituteFor("mysql"))
+        return MySQLContainer(imageName?.asCompatibleSubstituteFor("io/airbyte/integrations/destination/mysql"))
```
Comment (edgao) [inline airbyte-integrations/connectors/destination-mysql/src/testFixtures/kotlin/io/airbyte/integrations/destination/mysql/MysqlContainerFactory.kt:14]: what is this? afaict `mysql` is meant to be a docker image name, not a java package name?

**Review by edgao — COMMENTED** [review]
I didn't look at the actual java<>kotlin conversion, but had some questions about the other files

**Comment by stephane-airbyte** [comment]
* **#40553** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/40553?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @stephane-airbyte and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: airbyte-integrations/connectors/destination-mysql/src/testFixtures/kotlin/io/airbyte/integrations/destination/mysql/MysqlContainerFactory.kt (hunk 2 above)
Comment (stephane-airbyte) [reply airbyte-integrations/connectors/destination-mysql/src/testFixtures/kotlin/io/airbyte/integrations/destination/mysql/MysqlContainerFactory.kt:14]: so stupid refactoring coming from me moving files into the kotlin directory...

**Review by edgao — APPROVED** [review]

**Comment by stephane-airbyte** [comment]
### Merge activity

* **Jun 27, 4:44 PM EDT**: @stephane-airbyte started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/airbytehq/airbyte/40553).

**Pull request merged**